src/chain.h

```diff
@@ -320,6 +324,7 @@ class CDiskBlockIndex : public CBlockIndex
             READWRITE(VARINT(nDataPos));
         if (nStatus & BLOCK_HAVE_UNDO)
             READWRITE(VARINT(nUndoPos));
+        READWRITE(hashAnchor);
```

Comment:
We can't really detect it. The client should crash because there isn't enough data to read in each index.